### DISCUSSION TITLE
Does mouse wheel work for you in USB host mode ?

### DISCUSSION THREAD

**Discussion by shreekumar3d** [hathach/tinyusb#1714] [General]
I am using tinyusb on the RP2040 in host mode with mice.  I find that only with logitech mouse the mouse wheel value is correct (non-zero, positive and negative directions).  Mice of all other brands I have tested (Dell, and many other) all return 0.

Positing this here to check experience of others in the community...  Does the mouse wheel work for you ?

With a mouse I am testing now, I see that when I move the wheel, events are being generated.  I will be investigating this a bit more in detail in a few days, but meanwhile, any input will be most appreciated.

I have enabled logs (LOG=3).  I see that when mouse wheel is activated,  the data bytes are all 0 (same prints repeat every time I move the mouse)

> 
> Completed transfer of 3 bytes on ep 1 in
> on EP 81 with 3 bytes
> HID xfer callback
>   Get Report callback (2, 0)
>   0000:  00 00 00                                         |...|
> HID receive boot mouse report
> (0 0 0)
> 
> 

I find that strange, as I certainly expect the values in the report to change with the movement.

**Comment by shreekumar3d** [comment]
I switched the mouse to the report protocol (not boot).  Then each report was 5 bytes.  Looks like the report needs to be carefully parsed - only the first 3 bytes are supposed to be a bit standard.  The rest depend on the descriptor.

**Reply by hathach** [reply]
Yeah, currently host stack will set the protocol to BOOT mode since it currently lack an report parser. In the future this may change though when we have a decent parser.
